src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/dynamicProgramming/WrappingDPAlgorithm.java

```diff
@@ -0,0 +1,62 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.dynamicProgramming;
+
+import se.liu.ida.hefquin.engine.federation.FederationAccessException;
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.NaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.QueryOptimizationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WrappingDPAlgorithm {
+    protected final QueryOptimizationContext ctxt;
+
+    public WrappingDPAlgorithm( final QueryOptimizationContext ctxt ) {
+        assert ctxt != null;
+        this.ctxt = ctxt;
+    }
+
+    public PhysicalPlan optimize( final LogicalPlan initialPlan ) throws FederationAccessException {
```

Comment:
These methods here should not throw a FederationAccessException but a QueryOptimizationException.